lib/board.rb

```diff
@@ -0,0 +1,35 @@
+class Board
+  attr_accessor :field
+
+  def initialize
+    @field = [
+      1, 2, 3,
+      4, 5, 6,
+      7, 8, 9
+
+    ]
+  end
+
+  def show
+    puts "#{@field[0]} | #{@field[1]} | #{@field[2]}"
+    puts '--+---+---'
+    puts "#{@field[3]} | #{@field[4]} | #{@field[5]}"
+    puts '--+---+---'
+    puts "#{@field[6]} | #{@field[7]} | #{@field[8]}"
+    puts '--+---+---'
+  end
+
+  def full?
+    possible_values = %w[X O]
+    @field.all? { |cell| possible_values.include?(cell) }
+  end
+
+  def winner
+    winning_conditions = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [2, 4, 6], [0, 4, 8]]
+    winning_conditions.each do |condition|
+      return 'X' if condition.all? { |cell| @field[cell] == 'X' }
+      return 'O' if condition.all? { |cell| @field[cell] == 'O' }
+    end
+    false
+  end
```

Comment:
- DRY rule should be respected. E.g. students tend to check winning position for X and O separately instead of creating a method and reuse it. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/dry.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

```suggestion
  def winner(player_symbol)
    winning_conditions = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [2, 4, 6], [0, 4, 8]]
    winning_conditions.each do |condition|
      return player_symbol if condition.all? { |cell| @field[cell] == player_symbol }
    end
    false
  end
```